main.py

```diff
@@ -87,6 +90,12 @@ def post(self):
       else:
         uid = user['id']
 
+    # Check whether password is required
+    expected_password = self.settings['password']
+    if expected_password and \
+      self.get_argument('password') != expected_password:
```

Comment:
Shouldn't we use `hmac.compare_digest`?

Reply:
I didn't even consider the chance of an empowered adversary using timing tricks to target an image service, and I think it's more readable this way. But since you asked I'm fine with changing it